### PR TITLE
io/lab: Fix strtok_r usage in Mini-shell

### PR DESCRIPTION
**Pull request by teodorobert**
# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [x] Tested your changes against relevant architectures and platforms;
- [x] Updated relevant documentation (if needed).

## Description of changes

Currently the code fails to compile because `strtok_r` is not provided the `saveptr` argument.

This pull request fixes this and synchronizes the changes to the solution. It also removes the `TODO:` marker from one of the comments as there is nothing to do.

### CODE REVIEW

File: content/chapters/io/lab/solution/mini-shell/mini_shell.c [hunk 1]
```diff
@@ -140,6 +140,7 @@ static int parse_line(char *line)
 	int idx = 0;
 	char *token;
```
Comment (teodutu) [inline content/chapters/io/lab/solution/mini-shell/mini_shell.c:141]:
```suggestion
	char *token;
```

**Review by teodutu — CHANGES_REQUESTED** [review]
Looks legit. Thanks. Please remove the whitespace that I mentioned to pacify superlinter [1]. Do this within the same commit by using

```
git add <file_name>
git commit --amend --no-edit
git push -f
```

[1] https://github.com/cs-pub-ro/operating-systems/actions/runs/7173879807/job/19552630841?pr=42#step:3:621

**Comment by teodorobert** [comment]
Thanks. It should be good now.

**Review by teodutu — APPROVED** [review]
:ok_hand:

**Pull request merged**